.github/workflows/deploy-preview.yml

```
@@ -0,0 +1,131 @@
+name: GitHub Actions Rain Solver Preview Deployment
+
+on: [push]
+
+concurrency:
+  group: ${{ github.ref }}-deploy-preview
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+env:
+  CHANNEL: ${{ inputs.tag || github.head_ref || github.ref_name }}
```

actionlint (1.7.4)

property "tag" is not defined in object type {}

(expression)